addons/qodot/src/resources/game-definitions/fgd/qodot_fgd_model_point_class.gd

```diff
@@ -0,0 +1,105 @@
+@tool
+class_name QodotFGDModelPointClass
+extends QodotFGDPointClass
+
+## Optional - if empty, will use the game dir provided when exported.
+@export_global_dir var trenchbroom_game_dir = ""
+@export var model_export_dir := "trenchbroom/export"
+## Scale expression applied to model in Trenchbroom. See https://trenchbroom.github.io/manual/latest/#display-models-for-entities for more info.
+@export var scale_expression := ""
+@export var generate_bounding_box := true
+@export var apply_rotation_on_import := true
+@export var generate_gd_ignore_file := false
+func build_def_text() -> String:
+	_generate_model()
+	return super()
+
+func _generate_model():
+	_set_model()
+
+func _set_model():
+	if not scene_file:
+		return 
+	
+	var gltf_state := GLTFState.new()
+	var path = _get_export_dir()
+	var node = _get_node()
+	if node == null: return
+	if not _create_gltf_file(gltf_state, path, node, generate_gd_ignore_file):
+		printerr("could not create gltf file")
+		return
+	node.queue_free()
+	const model_key := "model"
+	const size_key := "size"
+	if scale_expression.is_empty():
+		meta_properties[model_key] = '"%s"' % _get_local_path()
+	else:
+		meta_properties[model_key] = '{"path": "%s", "scale": %s }' % [
+			_get_local_path(), 
+			scale_expression
+		]
+	if generate_bounding_box:
+		meta_properties[size_key] = _get_bounding_box(gltf_state.meshes)
+
+func _get_node() -> Node3D:
+	var node := scene_file.instantiate()
+	if node is Node3D: return node as Node3D
+	node.queue_free()
+	printerr("Scene is not of type 'Node3D'")
+	return null
+
+
+func _get_export_dir() -> String:
+	var tb_game_dir = QodotEditorSettings.get_trenchbroom_project_path() if trenchbroom_game_dir.is_empty() else trenchbroom_game_dir
+	print(tb_game_dir)
```

Comment:
You should remove this `print(tb_game_dir)` call so that it doesn't flood the Output if a user has a lot of these Model Point Classes in their FGD.